compiler/src/main/java/org/qbicc/graph/ReferenceHandle.java

```diff
@@ -18,10 +18,12 @@ public final class ReferenceHandle extends AbstractValueHandle {
     ReferenceHandle(Node callSite, ExecutableElement element, int line, int bci, Value referenceValue) {
         super(callSite, element, line, bci);
         this.referenceValue = referenceValue;
-        if (referenceValue.getType().isComplete()) {
-            pointerType = ((ReferenceType) referenceValue.getType()).getUpperBound().getPointer();
+        if (referenceValue.getType() instanceof ReferenceType rt) {
+            pointerType = rt.getUpperBound().getPointer();
+        } else if (referenceValue.getType().isUnresolved()) {
```

Comment:
Since there's only one `UnresolvedType`, I think it'd be better to use `instanceof` here rather than have an `isUnresolved()` which only returns `true` for one type.

Reply:
Fair enough.  Changed & squashed.